sys/linux/socket_netlink_generic_ethtool.txt

```diff
@@ -63,16 +88,83 @@ linkmodes_set_policy [
 	ETHTOOL_A_LINKMODES_OURS	nlnest[ETHTOOL_A_LINKMODES_OURS, array[bitset_policy]]
 ] [varlen]
 
-bitset_policy [
-	ETHTOOL_A_BITSET_NOMASK	nlattr[ETHTOOL_A_BITSET_NOMASK, void]
-	ETHTOOL_A_BITSET_SIZE	nlattr[ETHTOOL_A_BITSET_SIZE, int32]
-	ETHTOOL_A_BITSET_BITS	nlnest[ETHTOOL_A_BITSET_BITS, array[nlnest[ETHTOOL_A_BITSET_BITS_BIT, array[bit_policy]]]]
-	ETHTOOL_A_BITSET_VALUE	nlattr[ETHTOOL_A_BITSET_VALUE, array[int8]]
-	ETHTOOL_A_BITSET_MASK	nlattr[ETHTOOL_A_BITSET_MASK, array[int8]]
+debug_set_policy [
+	ETHTOOL_A_DEBUG_HEADER	nlnest[ETHTOOL_A_DEBUG_HEADER, array[ethnl_header_policy]]
+	ETHTOOL_A_DEBUG_MSGMASK	nlnest[ETHTOOL_A_DEBUG_MSGMASK, array[bitset_policy]]
 ] [varlen]
 
-bit_policy [
-	ETHTOOL_A_BITSET_BIT_INDEX	nlattr[ETHTOOL_A_BITSET_BIT_INDEX, int32]
-	ETHTOOL_A_BITSET_BIT_VALUE	nlattr[ETHTOOL_A_BITSET_BIT_VALUE, void]
-	ETHTOOL_A_BITSET_BIT_NAME	nlattr[ETHTOOL_A_BITSET_BIT_NAME, string]
+wol_set_policy [
+	ETHTOOL_A_WOL_HEADER	nlnest[ETHTOOL_A_WOL_HEADER, array[ethnl_header_policy]]
+	ETHTOOL_A_WOL_MODES	nlnest[ETHTOOL_A_WOL_MODES, array[bitset_policy]]
+	ETHTOOL_A_WOL_SOPASS	nlattr[ETHTOOL_A_WOL_SOPASS, array[int8]]
+] [varlen]
+
+features_set_policy [
+	ETHTOOL_A_FEATURES_HEADER	nlnest[ETHTOOL_A_FEATURES_HEADER, array[ethnl_header_policy]]
+	ETHTOOL_A_FEATURES_HW		nlnest[ETHTOOL_A_FEATURES_HW, array[bitset_policy]]
+	ETHTOOL_A_FEATURES_WANTED	nlnest[ETHTOOL_A_FEATURES_WANTED, array[bitset_policy]]
+	ETHTOOL_A_FEATURES_ACTIVE	nlnest[ETHTOOL_A_FEATURES_ACTIVE, array[bitset_policy]]
+	ETHTOOL_A_FEATURES_NOCHANGE	nlnest[ETHTOOL_A_FEATURES_NOCHANGE, array[bitset_policy]]
```

Comment:
I've run syz-check on this and it says:
```
+bad-netlink-attr: features_set_policy.ETHTOOL_A_FEATURES_HW: NLA_REJECT attribute will always be rejected
+bad-netlink-attr: features_set_policy.ETHTOOL_A_FEATURES_ACTIVE: NLA_REJECT attribute will always be rejected
+bad-netlink-attr: features_set_policy.ETHTOOL_A_FEATURES_NOCHANGE: NLA_REJECT attribute will always be rejected
```

I think these should not be listed here (not inputs).

Reply:
Thanks for noting that -- you are right, these are not included in the policy.

[Fixed](https://github.com/google/syzkaller/pull/1903/files#diff-4277f1131ebd61c5de5aff5107a42d09R102-R105).